src/main/java/ch/wisv/areafiftylan/web/banner/service/BannerServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package ch.wisv.areafiftylan.web.banner.service;
+
+import ch.wisv.areafiftylan.exception.BannerNotFoundException;
+import ch.wisv.areafiftylan.web.banner.model.Banner;
+import org.springframework.stereotype.Service;
+
+import java.sql.Date;
+import java.time.LocalDate;
+import java.util.Collection;
+
+@Service
+public class BannerServiceImpl implements BannerService {
+
+    private BannerRepository bannerRepository;
+
+    public BannerServiceImpl(BannerRepository bannerRepository) {
+        this.bannerRepository = bannerRepository;
+    }
+
+    @Override
+    public Collection<Banner> getAllBanners() {
+        return bannerRepository.findAll();
+    }
+
+    @Override
+    public Banner getCurrentbanner() {
+        Date now = Date.valueOf(LocalDate.now());
+        return bannerRepository.findFirstByStartDateLessThanEqualAndEndDateGreaterThanEqualOrderByIdDesc(now, now)
+                .orElseThrow(BannerNotFoundException::new);
+    }
+
+    @Override
+    public Banner addBanner(Banner banner) {
+        return bannerRepository.save(banner);
+    }
+
+    @Override
+    public Banner update(Long bannerId, Banner banner) {
+        if (!bannerRepository.exists(bannerId))
```

Comment:
if without `{ }` 😱

Reply:
It does work if it's a single line, you know that right :stuck_out_tongue:

Reply:
It works, but it's bad practice. There's no reason not to do it, and it prevents [nasty bugs](https://www.imperialviolet.org/2014/02/22/applebug.html)

Reply:
Yeah it is a bit nasty yeah